### PR TITLE
Integrate OAuth authentication custom claim check

### PR DESCRIPTION
**Pull request by mstruk**
### Type of change

- Enhancement / new feature

### Description

Adds ability to add custom validation of JWT token during authentication by specifying a JsonPath filter query to pass or fail the access token based on its content. If introspection endpoint is used, the filter query is applied to the response of the introspection endpoint. 

The following configuration option is added to the 'authentication' section of type 'oauth' of the Kafka CR:

```
  authentication:
    type: oauth
    customClaimCheck: "FILTER QUERY"
```

This PR depends on https://github.com/strimzi/strimzi-kafka-oauth/pull/87.
This PR is based on top of https://github.com/strimzi/strimzi-kafka-operator/pull/4027.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc [hunk 1]
```diff
@@ -135,6 +136,7 @@ Depending on how you apply OAuth 2.0 authentication, and the type of authorizati
...
 <8> Set this to `true` in order to enable the SASL_PLAIN mechanism on the listener, which is supported by all clients on all platforms. The Kafka client has to enable the SASL_PLAIN mechanism and set the `username` and the `password`. This mechanism can be used to authenticate either by using the OAuth access token, or by using the OAuth clientId and secret (client credentials). If the client sets `username` to `access-token` then the password is interpreted as access token on the server, otherwise the user is interpreted as the clientId, and password as the client secret. Default is `false`.
 <9> This has to be set to support the client credentials authentication when `enablePlain` is set to true, as described in previous point.
+<10> Additional custom rules can be imposed on the JWT access token during validation by setting this to a JsonPath filter query. If matching fails, the access token is rejected. When using `introspectionEndpointUri` the custom check is applied to the introspection endpoint response JSON.
```
Comment (PaulRMellor) [inline documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc:139]:
```suggestion
<10> Additional custom rules can be imposed on the JWT access token during validation by setting this to a JsonPath filter query. If the access token does not contain the necessary data, it is rejected. When using the `introspectionEndpointUri`, the custom check is applied to the introspection endpoint response JSON.
```

**Review by PaulRMellor — APPROVED** [review]
One small suggestion

**Review by ppatierno — APPROVED** [review]

**Pull request merged**